tests/utils/test_feature_utils.py

```diff
@@ -21,3 +21,109 @@ def test_normalize_coordinates(self):
+
+        non_homogenous_coordinates = np.array([])
+        computed = feature_utils.convert_to_homogenous_coordinates(non_homogenous_coordinates)
+
+        self.assertEqual(computed.size, 0)
+
+    def test_convert_to_homogenous_coordinates_on_none_input(self):
+        """Tests conversion to homogenous coordinates on None input."""
+
+        non_homogenous_coordinates = None
+        computed = feature_utils.convert_to_homogenous_coordinates(non_homogenous_coordinates)
+
+        self.assertIsNone(computed)
+
+    def test_convert_to_homogenous_coordinates_incorrect_dim(self):
+        """Tests conversion to homogenous coordinates on input which is already
+        3 dimensional."""
+
+        # test on invalid input
+        non_homogenous_coordinates = np.random.rand(5, 3)
+
+        self.assertRaises(
+            TypeError,
+            feature_utils.convert_to_homogenous_coordinates,
+            non_homogenous_coordinates,
+        )
+
+    def test_convert_to_epipolar_lines_valid_input(self):
+        """Test conversion of valid 2D points to epipolar lines using the essential matrix."""
+
+        points = np.array(
+            [
+                [10.0, -5.0],
+                [3.5, 20.0],
+            ]
+        )  # 2d points in homogenous coordinates
+        essential_mat = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
+
+        computed = feature_utils.convert_to_epipolar_lines(points, essential_mat)
+
+        expected = np.array(
+            [
+                [-2.36351579, -5.90878948, 1.75364193],
+                [-0.65653216, -1.64133041, -19.75129171],
+            ]
+        )
+
+        np.testing.assert_allclose(computed, expected)
+
+    def test_convert_to_epipolar_lines_empty_input(self):
+        """Test conversion of 0 2D points to epipolar lines using the essential matrix."""
+
+        points = np.array([])  # 2d points in homogenous coordinates
+        essential_mat = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
+
+        computed = feature_utils.convert_to_epipolar_lines(points, essential_mat)
+
+        self.assertEqual(computed.size, 0)
+
+    def test_convert_to_epipolar_lines_none_input(self):
+        """Test conversion of None to epipolar lines using the essential matrix."""
+
+        points = None
+        essential_mat = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
+
+        computed = feature_utils.convert_to_epipolar_lines(points, essential_mat)
+
+        self.assertIsNone(computed)
+
+    def test_compute_point_line_distances(self):
+        """Test for 2D point-line distance computation."""
+
+        points = np.array(
+            [
+                [-2.0, 1.0],
+                [5.0, 1.0],
+                [2.0, 3.0],
+            ]
+        )
+        lines = np.array(  # coefficients (a, b, c) for the line ax + by + cz = 0
```

Comment:
there is no z here right? just ax + by + c = 0?

Reply:
yes thats right. Thanks!